docs/resources/tls_subscription.md

```diff
@@ -71,24 +78,42 @@ resource "fastly_tls_subscription" "example" {
   certificate_authority = "lets-encrypt"
 }
 
-data "aws_route53_zone" "demo" {
-  name         = local.domain_name
-  private_zone = false
-}
-
 # Set up DNS record for managed DNS domain validation method
 resource "aws_route53_record" "domain_validation" {
   depends_on = [fastly_tls_subscription.example]
 
-  for_each        = {
-    for challenge in fastly_tls_subscription.example.managed_dns_challenges :
-    trimprefix(challenge.record_name, "_acme-challenge.") => challenge
+  # NOTE: in this example, two domains are added to the cert ("example.com" and "*.example.com").
+  # The "managed_dns_challenges" read-only attribute only includes one object
+  # for "_acme-challenge.example.com" as the challenge record is common in these two domains.
+  # 
+  # In order to process a cert contains wildcard entries, remove wildcard prefix "*." from the key
+  # and use ellipsis (...) to group results by key to avoid "Duplicate object key" error.
+  # Therefore, a key may have multiple elements. For example, domains "example.com" and "*.example.com"
+  # find the exact same object in the "managed_dns_challenges" attribute due to the "if" statement below.
+  # 
+  # A simplified version of this complex "for_each" usage would be:
+  # ```
+  # for_each = {
+  #   for challenge in fastly_tls_subscription.example.managed_dns_challenges :
+  #   trimprefix(challenge.record_name, "_acme-challenge.") => challenge
+  # }
+  # ```
+  # but since the "managed_dns_challenges" attribute is only known after apply,
+  # you will need to create this resource separately ("--target" option) and may not be ideal.
```

Comment:
Is it worth elaborating on the workflow the user would need to take? Unless it's just a case of the user running `terraform plan` to identify the changes and then `terraform apply -target="aws_route53_record.domain_validation"` and then `terraform apply` again if they have other things they want to be applied?

Reply:
@Integralist Thanks. I would leave it as is since the exact workflow may be varied depending upon users' config structure, strategy, etc. I think the current note suggests enough as to what they need if they really want to go with that route (also, TF error messages tell the same information nicely anyway).